ioSPI/cryoemio.py

```diff
@@ -0,0 +1,362 @@
+                                         added to the nominal defocus and the
+                                         systematic error, measured in um. A new
+                                         value of error is computed for every image.
+    - optics_defocusout [OPTIONAL] : if present, defocus values written to file.
+
+    *** detector_parameters ***
+    - detector_Nx                  : number of pixels on detector along x axis
+    - detector_Ny                  : number of pixels on detector along y axis
+    - detector_pixel_size          : physical pixel size in um
+    - detector_gain                : detector gain: avg number of counts per electron
+    - noise                        : quantized electron waves result in noise
+    - detector_Q_efficiency        : detector quantum efficiency
+    - MTF_params                   : list of 5 MTF parameters
+
+    *** miscellaneous ***
+    - seed [OPTIONAL]              : seed for the run. If not present, random.
+    """
+    parameters = None
+    with open(yaml_file, "r") as f:
+        parameters = yaml.safe_load(f)
+
+    # fill the dictionary
+    dic = {"simulation": {}}
+    if "miscellaneous" in parameters:
+        if "seed" in parameters["miscellaneous"]:
+            dic["simulation"]["seed"] = parameters["miscellaneous"]["seed"]
+        else:
+            random.seed()
+            dic["simulation"]["seed"] = random.randint(0, int(1e10))
+    else:
+        random.seed()
+        dic["simulation"]["seed"] = random.randint(0, int(1e10))
+    dic["simulation"]["log_file"] = log_file
+    dic["sample"] = {}
+    dic["sample"]["diameter"] = parameters["specimen_grid_params"][
+        "hole_diameter"
+    ]  # diameter in nm
+    dic["sample"]["thickness_center"] = parameters["specimen_grid_params"][
+        "hole_thickness_center"
+    ]  # thickness at center in nm
+    dic["sample"]["thickness_edge"] = parameters["specimen_grid_params"][
+        "hole_thickness_edge"
+    ]  # thickness at edge in nm
+    dic["particle"] = {}
+    dic["particle"]["name"] = parameters["molecular_model"]["particle_name"]
+    dic["particle"]["voxel_size"] = parameters["molecular_model"]["voxel_size"]
+    dic["particle"]["pdb_file"] = pdb_file
+    if "particle_mrcout" in parameters["molecular_model"]:
+        key = parameters["molecular_model"]["particle_mrcout"].split(".mrc")[0]
+        dic["particle"]["map_file_re_out"] = key + "_real.mrc"
+        dic["particle"]["map_file_im_out"] = key + "_imag.mrc"
+    else:
+        dic["particle"]["map_file_re_out"] = None
+    dic["particleset"] = {}
+    dic["particleset"]["name"] = parameters["molecular_model"]["particle_name"]
+    dic["particleset"]["crd_file"] = crd_file
+    dic["beam"] = {}
+    dic["beam"]["voltage"] = parameters["beam_parameters"]["voltage"]  # voltage in kV
+    dic["beam"]["spread"] = parameters["beam_parameters"][
+        "energy_spread"
+    ]  # energy spread in V
```

Comment:
put these comments in the code somehow, by putting units in variable names. comments are not good to maintain